deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -305,120 +306,123 @@ def running_simulations(self) -> Dict[str, List[str]]:
+            jobnames = [slot.get_job_name() for slot in sim_slots]
+            all_jobs_completed = all([(job in prior_completed_jobs) for job in jobnames])
+
+            self.instance_logger(f"jobnames: {jobnames}", debug=True)
+            self.instance_logger(f"All jobs completed?: {all_jobs_completed}", debug=True)
+
+            if all_jobs_completed:
+                do_terminate()
 
-            # first, figure out which jobs belong to this instance.
-            # if they are all completed already. RETURN, DON'T TRY TO DO ANYTHING
-            # ON THE INSTNACE.
-            parentslots = self.parent_node.sim_slots
-            rootLogger.debug("parentslots " + str(parentslots))
-            jobnames = [slot.get_job_name() for slot in parentslots if slot is not None]
-            rootLogger.debug("jobnames " + str(jobnames))
-            already_done = all([job in completed_jobs for job in jobnames])
-            rootLogger.debug("already done? " + str(already_done))
-            if already_done:
                 # in this case, all of the nodes jobs have already completed. do nothing.
                 # this can never happen in the cycle-accurate case at a point where we care
                 # about switch status, so don't bother to populate it
                 jobnames_to_completed = {jname: True for jname in jobnames}
-                return {'sims': jobnames_to_completed, 'switches': dict()}
+                return {'sims': jobnames_to_completed, 'switches': {}}
 
             # at this point, all jobs are NOT completed. so, see how they're doing now:
             instance_screen_status = self.running_simulations()
+
             switchescompleteddict = {k: False for k in instance_screen_status['switches']}
+            slotsrunning = [x for x in instance_screen_status['simdrivers']]
+            self.instance_logger(f"Switch Slots running: {switchescompleteddict}", debug=True)
+            self.instance_logger(f"Sim Slots running: {slotsrunning}", debug=True)
 
             if self.instance_assigned_switches():
-                # fill in whether switches have terminated for some reason
+                # fill in whether switches have terminated
                 for switchsim in self.parent_node.switch_slots:
-                    swname = switchsim.switch_builder.switch_binary_name()
-                    if swname not in switchescompleteddict.keys():
-                        switchescompleteddict[swname] = True
-
-            slotsrunning = [x for x in instance_screen_status['simdrivers']]
+                    sw_name = switchsim.switch_builder.switch_binary_name()
+                    if sw_name not in switchescompleteddict.keys():
+                        switchescompleteddict[sw_name] = True
 
-            rootLogger.debug("slots running")
-            rootLogger.debug(slotsrunning)
+            # fill in whether sims have terminated
+            completed_jobs = prior_completed_jobs.copy() # create local copy to append to
```

Comment:
The rest of this diff is all the same except for the `do_terminate` at the end (and some variable renames - again happy to revert).